### DISCUSSION TITLE
Test Procedures

### DISCUSSION THREAD

**Discussion by CarlosDerSeher** [CarlosDerSeher/snapclient#81] [Ideas]
_Originally posted by @CarlosDerSeher in https://github.com/CarlosDerSeher/snapclient/pull/69#issuecomment-2105188705_

Not sure what's the best thing to do. Maybe a ubuntu docker which integrates snapserver in some defined version. Automated tests of some form are not really possible here I guess as listening to actual audio will always be necessary doesn't it?
Continous Integration using github actions would be a big improvement, this way we can be sure there won't be any compile errors for different code configurations and changes won't break anything.

**Comment by DerPicknicker** [comment]
Hi @CarlosDerSeher , 

I really recommend this: [music assistant](https://music-assistant.io) for me it's the best solution for home audio. One command for setting everything up and play music on it. The only problem is that the buffer size cannot be adjusted until 2.1 release is out.

In general I would recommend to listen to music. You could easily listen if sync performance is poor or playback stucks. 

What do you think?

**Reply by CarlosDerSeher** [reply]
Seems to me there are a lot of variables involved if using music assistant in comparison to snapserver with some stream piped to the snapfifo

**Reply by DerPicknicker** [reply]
I would say yes and no. 

For a real world scenario it's better because more user will use that because it's an All in one solution. Setup is really easy. 

Do you have a NAS or a HomeServer? 

My current testings are looking like this: 
1. Use internal snapserver of MusicAssistant 
2. Use external Snapserver + MusicAssistant 

My planned option is that MA is my central music system. So maybe you try it out.

**Reply by CarlosDerSeher** [reply]
Ok I'll have a look when I find some time

**Reply by DerPicknicker** [reply]
If you need any assistance let me know

**Reply by DerPicknicker** [reply]
@CarlosDerSeher .. Did you found the time for testing MA?

**Reply by CarlosDerSeher** [reply]
No not yet

**Reply by CarlosDerSeher** [reply]
I've had another look at the feature list and it seems there isn't a smartphone app. How do you control a music assistant instance? Change streams, start stop playback, etc.

**Reply by DerPicknicker** [reply]
Just navigate to the adress of your HostIP:8095 .. navigate to settings turn on snapcast as player provider and you're done then you can integrate your music sources ..

The rest happens magically in the background.

**Reply by CarlosDerSeher** [reply]
Ok so all is controled through a browser UI ? I'll install and try out as soon as I can.

**Reply by DerPicknicker** [reply]
>  through a browser UI ? 

Exactly there is also a Desktop APP available, but it's not needed in my opinion because the WebUi contains all features. On Mobile you can add it to the home screen (android and iOS).. 
In the next Update (2.1 no target yet) .. You can create Airplay Speakers and Spotify Connect Speakers so then you can cast directly to your Snapcast-Speakers without going to MA. Which is kinda cool..
